### PR TITLE
GH-2930: Add issue templates

### PR DESCRIPTION
**Pull request by rok**
This introduces [Arrow-like issue templates](https://github.com/apache/arrow/tree/main/.github/ISSUE_TEMPLATE) to structure and categorize issues somewhat as they are reported.

See #2930

### CODE REVIEW

**Comment by rok** [comment]
@wgtmac

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (wgtmac) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Do we actually need these `options`? Or should it be consistent with modules like `parquet-avro`, `parquet-cli`, etc.? Same question for other yamls.

**Review by wgtmac — COMMENTED** [review]
Thanks for adding this!

cc @gszadovszky @julienledem @Fokko

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Also please note I introduced as many new labels as I could during migration. We now might want to reduce the current set by deleting them [here](https://github.com/apache/parquet-java/labels) (delete is irreversible). Same goes for [milestones](https://github.com/apache/parquet-java/milestones).

Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Thanks for the feedback @gszadovszky. I've change the list like so:
```diff
      options:
-       - Java
        - Release
-       - Documentation
        - Continuous Integration
-       - Packaging
+       - Arrow
+       - Avro
+       - Pig
+       - Protobuf
+       - Scala
+       - Thrift
+       - CLI
        - Benchmarking
        - Other
```

> I think it makes more sense to somehow sum up Release+CI+Packaging.

Agreed, but I don't have a good suggestion how. For now I just removed packaging.

Comment (gszadovszky) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Thank you, @rok!

It would be nice to see somehow how many issues we have related to `CI` and `Release`. Maybe `Other` is enough but not sure. I am fine as is, though.

One thing is clearly missing. This is the core functionality that most PRs have updates in. These are the modules `parquet-column`, `parquet-common`, `parquet-encoding`, `parquet-format-structures`, `parquet-generator`, `parquet-hadoop`. I am fine calling it `Core` if there are no better suggestion.

`parquet-hadoop` is definitely part of core currently, even though we want to get rid of the hadoop dependencies. After this work would be done, the separate component `Hadoop` would make sense, but currently I don't think we need it.

Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: To see distribution of component in Jira you can check the current (imported) label counts [here](https://github.com/apache/parquet-java/labels) - these are based on jira components, so CI and Release wont be present. For a quick idea of how much CI issues there are I made a quick dump of titles and bodies of issues: [issues_dump.csv](https://github.com/user-attachments/files/15985203/issues_dump.csv)

Comment (gszadovszky) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Some quick searches in the csv suggests we have enough release and CI related issues to have them separated. So, I'm fine with the list as is, only the `Core` label is missing.

Comment (Fokko) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Thanks @rok for working on this 🙌  I would be inclined to remove `Other` and make it optional. WDYT?

Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Thanks for the feedback! :D I've pushed a change and here's the current list:
```markdown
- Core
- Release
- Continuous Integration
- Arrow
- Avro
- Pig
- Protobuf
- Scala
- Thrift
- CLI
- Benchmarking
```
I'm glad to defer to your suggestions as I'm quite unfamiliar with `parquet-java` (except for the migration).

Comment (Fokko) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: One more thing, I was looking at Iceberg regarding @gszadovszky's comment.

> I think it makes more sense to somehow sum up Release+CI+Packaging.

At Iceberg we've packaged these into a `Build` label.

Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Seems like a good idea. Will do.

Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:35]: Done.

**Review by Fokko — APPROVED** [review]
LGTM, thanks @rok for working on this!

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (wgtmac) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:45]:
```suggestion
        - Benchmark
```

I would prefer `benchmark` here.

**Review by wgtmac — APPROVED** [review]
+1. Thanks @rok!

File: .github/ISSUE_TEMPLATE/bug_report.yaml
Comment (rok) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: Sure!

**Review by gszadovszky — APPROVED** [review]
Thanks, @rok!

**Comment by wgtmac** [comment]
I will merge this after next 24 hours if no objection.

File: .github/ISSUE_TEMPLATE/feature_request.yaml
Comment (wgtmac) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:48]:
```suggestion
        - Benchmark
```

File: .github/ISSUE_TEMPLATE/usage_question.yaml
Comment (wgtmac) [inline .github/ISSUE_TEMPLATE/usage_question.yaml:68]:
```suggestion
        - Benchmark
```

**Pull request merged**
